test/lib/Streamly/Test/Prelude/Common.hs

```diff
@@ -290,6 +293,41 @@ constructWithFromIndicesM op len =
         streamEffect <- run $ readIORef mvl
         listEquals (==) streamEffect list
 
+constructfromWAsync ::
+    S.WAsyncT IO Int -> S.WAsyncT IO Int-> [Int] -> Property
+constructfromWAsync s1 s2 res =
+    withMaxSuccess maxTestCount $
+    monadicIO $ do
+        x <-  run
+            $ S.toList
+            $ S.fromWAsync
+            $ S.maxThreads 1
+            $ s1 `S.wAsync` s2
+        equals (==) x res
+
+constructfromAsyncSingleThread ::
+    S.AsyncT IO Int -> S.AsyncT IO Int-> [Int] -> Property
+constructfromAsyncSingleThread s1 s2 res =
+    withMaxSuccess maxTestCount $
+    monadicIO $ do
+        x <-  run
+            $ S.toList
+            $ S.fromAsync
+            $ S.maxThreads 1
+            $ s1 `S.async` s2
+        equals (==) x res
```

Comment:
We can move this code to Async test module itself as this is specific to Async and not common.